TRAIN_AND_RUN_README.md

```diff
@@ -338,6 +338,70 @@ train_command \
   task_name=my_m2_training
 ```
 
+## The Global Step Predictor (GSP), and how it relates to the TCN Activity Classifier
+
+The above TCN activity classifier in its current configuration takes in a second or 
+two of video artifacts (e.g. for the "Locs&Confs" version, pose joint pixel coordinates 
+and confidences, the user's hand detection locations and confidences, and other procedure-relevant 
+object pixel coordinates and confidences), and outputs confidences for each of a vector 
+of activities (examples: "labels" in config/activity_labels/medical), assuming up to 
+one activity is ocurring "presently."
+
+Now, the GSP takes as input the confidence vector per frame window, and keeps track over time of which
+activities ("steps" in the GSP context) have occurred, and which step a user is on.
+
+Basically, if the "next step" at any point has been activated long enough, and with enough confidence, 
+the GSP progresses to that step as the latest "completed" step.
+
+Assumptions:
+- One activity or "background" (none of the listed activities) happens at a time.
+- The activities must happen in a specific linear order.
+- So, if an activity is detected with strong confidence way out of order (e.g. we're on step 3
+and we detect step 8), the GSP does not mark step 8 as completed.
+- A single "skipped step" is possible given some criteria. Skipping one step can be allowed unconditionally.
+Skipping one step can also be allowed if the "skipped step" has been activated with some "easier" criteria
+(a lower confidence threshold and/or fewer frames above that threshold). We can also configure to skip one step
+simply given that a threshold number of frames have passed since we completed the most recent step.
+
+## Training the GSP
```

Comment:
Nitpick: Don't need to add two high level sections (`##`'s). Is seems like the totality of the added content is about training the GSP, but it has two sub-sections (`###`'s): how it relates to the activity classifier and then the actual How-To with quick start.

Reply:
Made the two `##` sections into `###` subsections under one `##` section.